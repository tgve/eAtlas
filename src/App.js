/**
 * Alan Turing Geovisualization Engine (eAtlas)
 * 
 */
import React, { Component } from 'react';
import { Provider as StyletronProvider } from 'styletron-react';
import { BaseProvider, DarkTheme, LightTheme } from 'baseui';
import { Client as Styletron } from 'styletron-engine-atomic';

import Welcome from './Welcome';

import './App.css';
import { params } from './utils/api';

const engine = new Styletron();

export default function (props) {
  const apis = params(props, 
    props.location ? 
      props.location.search : window.location.search)
      
  return (
    <main>
      <StyletronProvider value={engine}>
        <BaseProvider 
          theme={apis.dark === false ? LightTheme : DarkTheme}>
          <Welcome
            {...apis}
          />
        </BaseProvider>
      </StyletronProvider>
    </main>
  )
};
