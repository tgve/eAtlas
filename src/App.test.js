import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { BaseProvider, DarkTheme, LightTheme } from 'baseui';

import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter><App /></BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('App - should create snapshot', () => {
  const component = renderer.create(
    <BrowserRouter><App /></BrowserRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('App - dark/light themes set', () => {
  const m = shallow(<App />);
  expect(m.find(BaseProvider).prop('theme')).toEqual(DarkTheme);

  const n = shallow(<App dark={false}/>);
  expect(n.find(BaseProvider).prop('theme')).toEqual(LightTheme);

})

test('App - API params set', () => {
  const m = shallow(<App />).find('Welcome');
  // console.log(m.find('Welcome').props());
  // console.log(m.debug());
  expect(m.props().hideCharts).toEqual(undefined);
  expect(m.props().hideChartGenerator).toEqual(undefined);
  expect(m.props().defaultURL).toEqual(undefined);

  const n = shallow(<App hideCharts={true}/>)
    .find('Welcome');
  expect(n.props().hideCharts).toEqual(true);

})

test('App - API params ENV',() => {
  process.env.REACT_APP_DEFAULT_URL = "https://react.com"
  const m = shallow(<App />).find('Welcome');
  expect(m.props().defaultURL).toEqual("https://react.com");
})