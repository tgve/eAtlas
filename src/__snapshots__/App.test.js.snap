// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App - dark/light themes set 1`] = `
<main>
  <div
    className=""
  >
    <div>
      <div
        className="loader"
        style={
          Object {
            "visibility": "visible",
            "zIndex": -1,
          }
        }
      />
      <div
        style={
          Object {
            "cursor": "grab",
            "height": "768px",
            "position": "relative",
            "width": "1024px",
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            className=""
            style={
              Object {
                "height": "100%",
                "overflow": "hidden",
                "position": "absolute",
                "visibility": "inherit",
                "width": "100%",
              }
            }
          />
          <div
            style={
              Object {
                "height": 0,
                "overflow": "visible",
                "width": 0,
              }
            }
          />
        </div>
      </div>
      <div
        className="side-panel-container"
        style={
          Object {
            "marginLeft": 0,
          }
        }
      >
        <div
          className="side-panel"
          style={
            Object {
              "background": "#242730",
              "color": "white",
            }
          }
        >
          <div
            className="ae af ag ah"
          >
            <h2>
              Nothing to show
            </h2>
          </div>
          <div>
            <button
              className="ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk bl ae bm bn"
              data-baseweb="button"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Add data
            </button>
            <i
              className="fa fa-save"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                  "margin": 5,
                }
              }
            />
          </div>
          <div
            className="side-panel-body"
          >
            <div
              className="side-panel-body-content"
            >
              <br />
              <hr
                style={
                  Object {
                    "clear": "both",
                  }
                }
              />
              <div
                className="bo"
              >
                <div
                  aria-orientation="horizontal"
                  className="bp bq br aj bs bt bu bv bw bx"
                  data-baseweb="tab-list"
                  role="tablist"
                >
                  <button
                    aria-controls="tabs-bui1-tabpanel-0"
                    aria-selected={true}
                    className="b1 ax b5 b7 b6 b8 by ai ak bz c0 c1 c2 am an ao ap aq ar as at bl c3 ay az b0 au c4 b2 c5 c6 c7 b9 ba bb bc al"
                    data-baseweb="tab"
                    disabled={false}
                    id="tabs-bui1-tab-0"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    role="tab"
                    tabIndex="0"
                    type="button"
                  >
                    <i
                      className="fa fa-info"
                      style={
                        Object {
                          "fontSize": "2rem",
                        }
                      }
                    />
                  </button>
                  <button
                    aria-controls="tabs-bui1-tabpanel-1"
                    aria-selected={false}
                    className="b1 ax b5 b7 b6 b8 by ai ak bz c0 c1 c2 am an ao ap aq ar as at bl c3 ay az b0 au c4 b2 c5 c6 c7 b9 ba bb bc al"
                    data-baseweb="tab"
                    disabled={false}
                    id="tabs-bui1-tab-1"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    role="tab"
                    tabIndex="-1"
                    type="button"
                  >
                    <i
                      className="fa fa-sliders"
                      style={
                        Object {
                          "fontSize": "2rem",
                        }
                      }
                    />
                  </button>
                  <div
                    aria-hidden="true"
                    className="c8 c9 ca cb cc cd ce cf"
                    data-baseweb="tab-highlight"
                    role="presentation"
                  />
                </div>
                <div
                  aria-hidden="true"
                  className="ae bp cd"
                  data-baseweb="tab-border"
                  role="presentation"
                />
                <div
                  aria-labelledby="tabs-bui1-tab-0"
                  className="cg au"
                  data-baseweb="tab-panel"
                  hidden={false}
                  id="tabs-bui1-tabpanel-0"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="tabs-bui1-tab-1"
                  className="cg au"
                  data-baseweb="tab-panel"
                  hidden={true}
                  id="tabs-bui1-tabpanel-1"
                  role="tabpanel"
                />
              </div>
            </div>
            <div
              className="space"
            />
            <form
              className="search-form"
              onSubmit={[Function]}
            >
              <div
                className="ch ci"
                data-baseweb="form-control-container"
              >
                <div
                  className="by bq cj ch ck cl cm cn co cp cq cr cs az ct bf bg bd be b9 cu cv cw cx cy cz d0 d1 d2 d3"
                  data-baseweb="input"
                >
                  <div
                    className="bq ch d4 az ct b9 cu cv cw bl d1"
                    data-baseweb="base-input"
                  >
                    <input
                      aria-activedescendant={null}
                      aria-autocomplete={null}
                      aria-controls={null}
                      aria-describedby={null}
                      aria-errormessage={null}
                      aria-haspopup={null}
                      aria-invalid={false}
                      aria-label={null}
                      aria-labelledby={null}
                      aria-required={false}
                      autoComplete="on"
                      className="by d5 am ao an ap aq as ar at au ch d6 d7 d8 d9 bh bi da d3 b9 cu cv cw bl db dc"
                      disabled={false}
                      id="search-nominatum"
                      inputMode="text"
                      name=""
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyPress={[Function]}
                      onKeyUp={[Function]}
                      pattern={null}
                      placeholder="fly to ..."
                      required={false}
                      role={null}
                      rows={null}
                      type="text"
                    />
                  </div>
                  <div
                    className="bq ak al dd az ct b9 cu cv cw de df bl d1"
                  >
                    üåê
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          className="close-button"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "#242730",
              }
            }
          >
            <i
              className="fa fa-arrow-circle-left"
              style={
                Object {
                  "color": "white !important",
                  "fontSize": "2rem",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className=""
  />
</main>
`;

exports[`App - dark/light themes set 2`] = `
<main>
  <div
    className=""
  >
    <div>
      <div
        className="loader"
        style={
          Object {
            "visibility": "visible",
            "zIndex": -1,
          }
        }
      />
      <div
        style={
          Object {
            "cursor": "grab",
            "height": "768px",
            "position": "relative",
            "width": "1024px",
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            className=""
            style={
              Object {
                "height": "100%",
                "overflow": "hidden",
                "position": "absolute",
                "visibility": "inherit",
                "width": "100%",
              }
            }
          />
          <div
            style={
              Object {
                "height": 0,
                "overflow": "visible",
                "width": 0,
              }
            }
          />
        </div>
      </div>
      <div
        className="side-panel-container"
        style={
          Object {
            "marginLeft": 0,
          }
        }
      >
        <div
          className="side-panel"
          style={
            Object {
              "background": "white",
              "color": "black",
            }
          }
        >
          <div
            className="dg af ag ah"
          >
            <h2>
              Nothing to show
            </h2>
          </div>
          <div>
            <button
              className="ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 dh di b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk dj dg dk dl"
              data-baseweb="button"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Add data
            </button>
            <i
              className="fa fa-save"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                  "margin": 5,
                }
              }
            />
          </div>
          <div
            className="side-panel-body"
          >
            <div
              className="side-panel-body-content"
            >
              <br />
              <hr
                style={
                  Object {
                    "clear": "both",
                  }
                }
              />
              <div
                className="bo"
              >
                <div
                  aria-orientation="horizontal"
                  className="bp bq br aj bs bt bu bv bw bx"
                  data-baseweb="tab-list"
                  role="tablist"
                >
                  <button
                    aria-controls="tabs-bui1-tabpanel-0"
                    aria-selected={true}
                    className="b1 ax b5 b7 b6 b8 by ai ak bz c0 c1 c2 am an ao ap aq ar as at dj dm ay az b0 au c4 b2 di dn c7 b9 ba bb bc al"
                    data-baseweb="tab"
                    disabled={false}
                    id="tabs-bui1-tab-0"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    role="tab"
                    tabIndex="0"
                    type="button"
                  >
                    <i
                      className="fa fa-info"
                      style={
                        Object {
                          "fontSize": "2rem",
                        }
                      }
                    />
                  </button>
                  <button
                    aria-controls="tabs-bui1-tabpanel-1"
                    aria-selected={false}
                    className="b1 ax b5 b7 b6 b8 by ai ak bz c0 c1 c2 am an ao ap aq ar as at dj dm ay az b0 au c4 b2 di dn c7 b9 ba bb bc al"
                    data-baseweb="tab"
                    disabled={false}
                    id="tabs-bui1-tab-1"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    role="tab"
                    tabIndex="-1"
                    type="button"
                  >
                    <i
                      className="fa fa-sliders"
                      style={
                        Object {
                          "fontSize": "2rem",
                        }
                      }
                    />
                  </button>
                  <div
                    aria-hidden="true"
                    className="do c9 ca cb cc cd ce cf"
                    data-baseweb="tab-highlight"
                    role="presentation"
                  />
                </div>
                <div
                  aria-hidden="true"
                  className="c8 bp cd"
                  data-baseweb="tab-border"
                  role="presentation"
                />
                <div
                  aria-labelledby="tabs-bui1-tab-0"
                  className="cg au"
                  data-baseweb="tab-panel"
                  hidden={false}
                  id="tabs-bui1-tabpanel-0"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="tabs-bui1-tab-1"
                  className="cg au"
                  data-baseweb="tab-panel"
                  hidden={true}
                  id="tabs-bui1-tabpanel-1"
                  role="tabpanel"
                />
              </div>
            </div>
            <div
              className="space"
            />
            <form
              className="search-form"
              onSubmit={[Function]}
            >
              <div
                className="ch ci"
                data-baseweb="form-control-container"
              >
                <div
                  className="by bq cj ch ck cl cm cn co cp cq cr cs az ct bf bg bd be b9 cu cv cw dp dq dr ds dg d2 d3"
                  data-baseweb="input"
                >
                  <div
                    className="bq ch d4 az ct b9 cu cv cw dj dg"
                    data-baseweb="base-input"
                  >
                    <input
                      aria-activedescendant={null}
                      aria-autocomplete={null}
                      aria-controls={null}
                      aria-describedby={null}
                      aria-errormessage={null}
                      aria-haspopup={null}
                      aria-invalid={false}
                      aria-label={null}
                      aria-labelledby={null}
                      aria-required={false}
                      autoComplete="on"
                      className="by d5 am ao an ap aq as ar at au ch d6 d7 d8 d9 bh bi da d3 b9 cu cv cw dj dt dc"
                      disabled={false}
                      id="search-nominatum"
                      inputMode="text"
                      name=""
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyPress={[Function]}
                      onKeyUp={[Function]}
                      pattern={null}
                      placeholder="fly to ..."
                      required={false}
                      role={null}
                      rows={null}
                      type="text"
                    />
                  </div>
                  <div
                    className="bq ak al dd az ct b9 cu cv cw de df dj dg"
                  >
                    üåê
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          className="close-button"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "#242730",
              }
            }
          >
            <i
              className="fa fa-arrow-circle-left"
              style={
                Object {
                  "color": "white !important",
                  "fontSize": "2rem",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className=""
  />
</main>
`;
